fe/fe-core/src/main/java/org/apache/doris/qe/cache/CacheAnalyzer.java

```diff
@@ -0,0 +1,450 @@
+            return CacheMode.None;
+        }
+        //Check if whereClause have one CompoundPredicate of partition column
+        List<CompoundPredicate> compoundPredicates = Lists.newArrayList();
+        getPartitionKeyFromSelectStmt(this.selectStmt, partColumn, compoundPredicates);
+        if (compoundPredicates.size() != 1) {
+            LOG.info("the predicate size include partition key has {}", compoundPredicates.size());
+            return CacheMode.None;
+        }
+        partitionPredicate = compoundPredicates.get(0);
+        cache = new PartitionCache(this.queryId, this.selectStmt);
+        ((PartitionCache) cache).setCacheInfo(this.latestTable, this.partitionInfo, this.partColumn,
+                this.partitionPredicate);
+        MetricRepo.COUNTER_CACHE_MODE_PARTITION.increase(1L);
+        return CacheMode.Partition;
+    }
+
+    public CacheBeProxy.FetchCacheResult getCacheData() {
+        CacheProxy.FetchCacheResult cacheResult = null;
+        cacheMode = innerCheckCacheMode(0);
+        if (cacheMode == CacheMode.NoNeed) {
+            return cacheResult;
+        }
+        if (cacheMode == CacheMode.None) {
+            LOG.info("check cache mode {}, queryid {}", cacheMode, DebugUtil.printId(queryId));
+            return cacheResult;
+        }
+        Status status = new Status();
+        cacheResult = cache.getCacheData(status);
+
+        if (status.ok() && cacheResult != null) {
+            LOG.info("hit cache, mode {}, queryid {}, all count {}, value count {}, row count {}, data size {}",
+                    cacheMode, DebugUtil.printId(queryId),
+                    cacheResult.all_count, cacheResult.value_count,
+                    cacheResult.row_count, cacheResult.data_size);
+        } else {
+            LOG.info("miss cache, mode {}, queryid {}, code {}, msg {}", cacheMode,
+                    DebugUtil.printId(queryId), status.getErrorCode(), status.getErrorMsg());
+            cacheResult = null;
+        }
+        return cacheResult;
+    }
+
+    public long nowtime() {
+        return System.currentTimeMillis();
+    }
+
+    private void getPartitionKeyFromSelectStmt(SelectStmt stmt, Column partColumn,
+                                               List<CompoundPredicate> compoundPredicates) {
+        getPartitionKeyFromWhereClause(stmt.getWhereClause(), partColumn, compoundPredicates);
+        List<TableRef> tableRefs = stmt.getTableRefs();
+        for (TableRef tblRef : tableRefs) {
+            if (tblRef instanceof InlineViewRef) {
+                InlineViewRef viewRef = (InlineViewRef) tblRef;
+                QueryStmt queryStmt = viewRef.getViewStmt();
+                if (queryStmt instanceof SelectStmt) {
+                    getPartitionKeyFromSelectStmt((SelectStmt) queryStmt, partColumn, compoundPredicates);
+                }
+            }
+        }
+    }
+
+    /**
+     * Only support case 1
+     * 1.key >= a and key <= b
+     * 2.key = a or key = b
```

Comment:
Why don't support key = a?  only one BinaryPredicate.

Reply:
Other cases are more complicated, I think the first version is simpler